### ISSUE TITLE
New package: xf86-video-r128, xf86-video-mach64

### ISSUE THREAD

**Issue by q66** [void-ppc/void-packages#31]
**Status:** initial template created and building (patching necessary), but segfaults on start - needs to be investigated on actual hardware, which i don't have

We should also package `xf86-video-mach64`, which is for older Rage/Rage Pro cards, but there I don't even have anybody totest that, so it'll wait until requested.

This is to support at least 2D acceleration on old 32-bit Macs with ATi Rage 128 series GPUs, instead of plain `fbdev`. Matching ~~DRM~~framebuffer kernel driver should already be supplied.

**Comment by HadetTheUndying** [comment]
I'm not savvy enough to package this on my own but I can definitely help you text it when you're ready.

**Comment by cunotj** [comment]
I have an Imac G3 500. Let me know what I can do to help.

**Comment by q66** [comment]
Well, you can try debugging/fixing the crash :P Here's the template with patches if you wanna try your hand at it https://gist.github.com/q66/e48a590420a8ad0256561e8f70ae0661

Just testing alone won't really do, this needs someone who can find their way around the problem (or provide me with access to affected hardware so I can do it myself)

**Comment by HadetTheUndying** [comment]
I'll have free time next week. I got out my old C book to refresh my brain.

**Comment by q66** [comment]
@HadetTheUndying i have a new patch, can you test it out? https://gist.github.com/q66/f33d8d9afa2a685b20f938c3254750dd

**Comment by q66** [comment]
Anyone else (@cunotj etc) feel free to test this as well, if you have Void running on affected hardware.

I'd also like to know whether the `r128` kernel driver is loaded by default (listing from `lsmod`) and whether blacklisting it (`modprobe.blacklist=r128` on kernel command line, e.g. via `/etc/default/grub`) has any effect on whether `xf86-video-fbdev` works; from our previous testing it did not work as-is (only after adding `video=aty128fb:off` on kernel and setting xorg to 8 bit colors; on the default framebuffer it would report "No screens found").

**Comment by cunotj** [comment]
@q66 I have not tested anything yet, but 'aty128fb: ROM failed to map' is the first thing i see on boot.
Kernel driver is not listed in `lsmod`.

**Comment by HadetTheUndying** [comment]
@cunotj a couple of weeks ago when @q66 was helping me with this we were able to get a display environment working in 256-color by using the OpenFirmware FB, but obviously that's not ideal. I'll be around after Saturday to help test some things.

**Comment by q66** [comment]
both are now packaged and should be patched to work (r128 needs manual xorg.conf.d at least on powerpc), if there are any issues they can be made into a separate ticket
